api/metrics/src/main/java/io/opentelemetry/api/metrics/DoubleGaugeBuilder.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.api.metrics;
+
+import java.util.function.Consumer;
+
+/** A builder for Gauge metric types. These can only be asynchronously collected. */
+public interface DoubleGaugeBuilder {
+  /**
+   * Sets the description for this instrument.
+   *
+   * <p>Description strings should follow the instrument description rules:
+   * https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#instrument-description
+   */
+  DoubleGaugeBuilder setDescription(String description);
+
+  /**
+   * Sets the unit of measure for this instrument.
+   *
+   * <p>Unit strings should follow the instrument unit rules:
+   * https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#instrument-unit
+   */
+  DoubleGaugeBuilder setUnit(String unit);
+
+  /** Sets the gauge for recording {@code long} values. */
+  LongGaugeBuilder ofLongs();
+
+  /**
+   * Builds this asynchronous instrument with the given callback.
+   *
+   * <p>The callback will only be called when the {@link Meter} is being observed.
+   *
+   * @param callback A state-capturing callback used to observe values on-demand.
+   */
+  void buildWithCallback(Consumer<ObservableDoubleMeasurement> callback);
```

Comment:
Personally I don't like the mix of sync/async instruments in the same builder. May lead to confusions. I understand that the downside is some duplicate code, but I don't think is a big deal.

Reply:
FWIU of the discussions, the user feedback is that it's complicated to understand the difference between a sync and async instrument, they record the same thing just one has a callback for retrieving values instead of direct add. There is no change in the actual data emitted by the SDK. So mixing means you first pick 1) what you want to record and then 2) how you want to record it rather than combining those into one thought process, which to me at least does seem confusing.

Reply:
That being said, perhaps it gets simplified if gauge was the only asynchronous instrument. I'm sure that's been discussed though I haven't been following the spec discussions. But intuitively indeed I'm not too sure when I would use a callback with an updowncounter for example.

Reply:
The switch to merge these together was an ask in the Java SiG meeting.   I agree with @anuraaga that picking the type of metric is likely to be primary concern.    Would be good if you could join the Java SiG (when @jkwatson is back from vacation) to discuss this specific choice.  I disagree on confusion as I think the opposite is true: Splitting these leads to greater confusion.   If this was purely about duplicated code, I agree it's not a big deal.   This design was chosen to simplify and reduce user confusion.

@anuraaga I think, likely, the Java instrumentation folk may need to be using UpDownCounter more often.    E.g. JVM memory usage SHOULD be an up-down-counter callback.

Reply:
@jsuereth Please ping me when to join, would be happy to do it.

Reply:
Thursday at 9am PDT, next week should have @jkwatson back, I'll be unable to make it but I'm ok accepting whatever you hash out there.  Alternatively, may be worth joining the 4pm PDT meeting (Thursday august 5th) w/ @jkwatson + @anuraaga for high bandwidth discussion.

Reply:
My opinion is that this is a good limitation of the API surface, but I can also see the point that it could be confusing to users who are new to the API. Hard call. I *personally* like this API better than having separate builders, but I'm only one user who is too deep in the weeds to know if this would be confusing. 

@kenfinnigan Any thoughts?

Reply:
wait... this particular case isn't a mix of sync/async is it? It's just async, since the gauge is only async, correct?